paasta_tools/paasta_metastatus.py

```diff
@@ -286,11 +285,36 @@ def print_results_for_healthchecks(summary, ok, results, verbose):
             print_with_indent(line, 2)
 
 
+def get_extra_mesos_slave_data(mesos_state):
+    slaves = dict((slave['id'], slave) for slave in mesos_state['slaves'])
```

Comment:
why do you need the whole slave dict as the value here? how about:

```
dict((slave['id'], {}) for slave in mesos_state['slaves'])
```

that way, in the below logic, you're won't be modifying the slave dict.